### PR TITLE
Wrapping of embedded statements

### PR DESCRIPTION
**Pull request by Pilchie**

### CODE REVIEW

**Comment by Pilchie** [comment]
Tagging @dotnet/mlangide as well as @basoundr and @heejaechang

File: src/Workspaces/CSharp/Portable/Extensions/SyntaxNodeExtensions.cs [hunk 1]
```diff
@@ -960,25 +960,33 @@ public static SyntaxNode GetParent(this SyntaxNode node)
...
-            return node is IfStatementSyntax ||
+            return
+                   node is DoStatementSyntax ||
                    node is ElseClauseSyntax ||
-                   node is WhileStatementSyntax ||
-                   node is ForStatementSyntax ||
+                   node is FixedStatementSyntax ||
                    node is ForEachStatementSyntax ||
+                   node is ForStatementSyntax ||
+                   node is IfStatementSyntax ||
```
Comment (basoundr) [inline src/Workspaces/CSharp/Portable/Extensions/SyntaxNodeExtensions.cs]: Some tests for the newly introduced syntax?

File: src/Workspaces/CSharp/Portable/Formatting/Rules/WrappingFormattingRule.cs [hunk 2]
```diff
@@ -35,55 +36,59 @@ public override void AddSuppressOperations(List<SuppressOperation> list, SyntaxN
...
             {
-                AddSuppressWrappingIfOnSingleLineOperation(list, ifStatementNode.IfKeyword, ifStatementNode.Statement.GetLastToken(includeZeroWidth: true));
-
-                if (ifStatementNode.Else != null)
+                var firstTokenOfEmbeddedStatement = embeddedStatement.GetFirstToken(includeZeroWidth: true);
+                if (embeddedStatement.IsKind(SyntaxKind.Block))
                 {
-                    AddSuppressWrappingIfOnSingleLineOperation(list, ifStatementNode.Else.ElseKeyword, ifStatementNode.Else.Statement.GetLastToken(includeZeroWidth: true));
+                    return ValueTuple.Create(
+                        firstTokenOfEmbeddedStatement.GetPreviousToken(includeZeroWidth: true),
```
Comment (basoundr) [inline src/Workspaces/CSharp/Portable/Formatting/Rules/WrappingFormattingRule.cs]: extract `firstTokenOfEmbeddedStatement.GetPreviousToken(includeZeroWidth: true)`out of `if`

File: src/Workspaces/CSharp/Portable/Formatting/Rules/WrappingFormattingRule.cs [hunk 3]
```diff
@@ -112,46 +117,16 @@ private void RemoveSuppressOperationForStatementMethodDeclaration(List<SuppressO
...
             }
 
-            var anonymousMethod = node as AnonymousMethodExpressionSyntax;
-            if (anonymousMethod != null)
+            var ifStatementNode = node as IfStatementSyntax;
+            if (ifStatementNode?.Else != null)
             {
-                RemoveSuppressOperation(list, anonymousMethod.DelegateKeyword, anonymousMethod.GetLastToken(includeZeroWidth: true));
-                return;
+                RemoveSuppressOperation(list, ifStatementNode.Else.ElseKeyword, ifStatementNode.Else.Statement.GetFirstToken(includeZeroWidth: true));
```
Comment (basoundr) [inline src/Workspaces/CSharp/Portable/Formatting/Rules/WrappingFormattingRule.cs]: Is there a testcase for changing the `GetLastToken` to `GetFirstToken`

File: src/Workspaces/CSharp/Portable/Formatting/Rules/WrappingFormattingRule.cs (hunk 2 above)
Comment (DustinCampbell) [inline src/Workspaces/CSharp/Portable/Formatting/Rules/WrappingFormattingRule.cs]: I'd actually push the logic into conditional expression:

```C#
var firstTokenOfEmbeddedStatement = embeddedStatement.GetFirstToken(includeZeroWidth: true);
var tokenBeforeEmbeddedStatement = firstTokenOfEmbeddedStatement.GetPreviousToken(includeZeroWidth: true);
return ValueTuple.Create(
    tokenBeforeEmbeddedStatement,
    embeddingStatement.IsKind(SyntaxKind.Block)
        ? embeddedStatement.GetLastToken(includeZeroWidth: true)
        : firstTokenOfEmbeddedStatement));
```

Not a big deal though.

**Comment by DustinCampbell** [comment]
:+1:

**Pull request merged**